parser/tools/generate_ast.cc

```diff
@@ -305,7 +305,7 @@ NodeDef nodes[] = {
     {
         "Hash",
         "hash",
-        vector<FieldDef>({{"kwargs", FieldType::Bool}, {"pairs", FieldType::NodeVec}}),
+        vector<FieldDef>({{"inline_", FieldType::Bool}, {"pairs", FieldType::NodeVec}}),
```

Comment:
As this boolean would be true for cases like `foo(10 => 20)` as well, it seemed easier to rename it rather than special case symbol keys in the parser.

Reply:
worth noting that the boolean is in fact called `kwargs` in the Ruby VM iirc

Reply:
```
❯ ruby --dump=parsetree -e 'foo({x: 0})'
###########################################################
## Do NOT use this node dump for any purpose other than  ##
## debug and research.  Compatibility is not guaranteed. ##
###########################################################

# @ NODE_SCOPE (line: 1, location: (1,0)-(1,11))
# +- nd_tbl: (empty)
# +- nd_args:
# |   (null node)
# +- nd_body:
#     @ NODE_FCALL (line: 1, location: (1,0)-(1,11))*
#     +- nd_mid: :foo
#     +- nd_args:
#         @ NODE_ARRAY (line: 1, location: (1,4)-(1,10))
#         +- nd_alen: 1
#         +- nd_head:
#         |   @ NODE_HASH (line: 1, location: (1,4)-(1,10))
#         |   +- nd_alen: 1 (hash literal)
#         |   +- nd_head:
#         |       @ NODE_ARRAY (line: 1, location: (1,5)-(1,9))
#         |       +- nd_alen: 2
#         |       +- nd_head:
#         |       |   @ NODE_LIT (line: 1, location: (1,5)-(1,7))
#         |       |   +- nd_lit: :x
#         |       +- nd_head:
#         |       |   @ NODE_LIT (line: 1, location: (1,8)-(1,9))
#         |       |   +- nd_lit: 0
#         |       +- nd_next:
#         |           (null node)
#         +- nd_next:
#             (null node)

~/sorbet (master)
❯ ruby --dump=parsetree -e 'foo(x: 0)'
###########################################################
## Do NOT use this node dump for any purpose other than  ##
## debug and research.  Compatibility is not guaranteed. ##
###########################################################

# @ NODE_SCOPE (line: 1, location: (1,0)-(1,9))
# +- nd_tbl: (empty)
# +- nd_args:
# |   (null node)
# +- nd_body:
#     @ NODE_FCALL (line: 1, location: (1,0)-(1,9))*
#     +- nd_mid: :foo
#     +- nd_args:
#         @ NODE_ARRAY (line: 1, location: (1,4)-(1,8))
#         +- nd_alen: 1
#         +- nd_head:
#         |   @ NODE_HASH (line: 1, location: (1,4)-(1,8))
#         |   +- nd_alen: 0 (keyword argument)
#         |   +- nd_head:
#         |       @ NODE_ARRAY (line: 1, location: (1,4)-(1,8))
#         |       +- nd_alen: 2
#         |       +- nd_head:
#         |       |   @ NODE_LIT (line: 1, location: (1,4)-(1,6))
#         |       |   +- nd_lit: :x
#         |       +- nd_head:
#         |       |   @ NODE_LIT (line: 1, location: (1,7)-(1,8))
#         |       |   +- nd_lit: 0
#         |       +- nd_next:
#         |           (null node)
#         +- nd_next:
#             (null node)

~/sorbet (master)
❯ ruby --dump=parsetree -e 'foo(10 => 20)'
###########################################################
## Do NOT use this node dump for any purpose other than  ##
## debug and research.  Compatibility is not guaranteed. ##
###########################################################

# @ NODE_SCOPE (line: 1, location: (1,0)-(1,13))
# +- nd_tbl: (empty)
# +- nd_args:
# |   (null node)
# +- nd_body:
#     @ NODE_FCALL (line: 1, location: (1,0)-(1,13))*
#     +- nd_mid: :foo
#     +- nd_args:
#         @ NODE_ARRAY (line: 1, location: (1,4)-(1,12))
#         +- nd_alen: 1
#         +- nd_head:
#         |   @ NODE_HASH (line: 1, location: (1,4)-(1,12))
#         |   +- nd_alen: 0 (keyword argument)
#         |   +- nd_head:
#         |       @ NODE_ARRAY (line: 1, location: (1,4)-(1,12))
#         |       +- nd_alen: 2
#         |       +- nd_head:
#         |       |   @ NODE_LIT (line: 1, location: (1,4)-(1,6))
#         |       |   +- nd_lit: 10
#         |       +- nd_head:
#         |       |   @ NODE_LIT (line: 1, location: (1,10)-(1,12))
#         |       |   +- nd_lit: 20
#         |       +- nd_next:
#         |           (null node)
#         +- nd_next:
#             (null node)
```

and in fact, it looks like `keyword argument` is `0` from the parser on the `10 => 20` case, so there might be an argument in favor of fixing this in the `parser/Builder.cc` code